fhirpipe/references/reference_binder.py

```diff
@@ -0,0 +1,123 @@
+import re
+import logging
+from collections import defaultdict
+
+import fhirpipe
+from fhirpipe.load.fhirstore import get_mongo_client, get_resource_instances
+
+
+class ReferenceBinder:
+    def __init__(self, fhirstore, skip_ref_binding=False):
+        self.fhirstore = fhirstore
+        self.skip_ref_binding = skip_ref_binding
+
+        # store of the form
+        # {
+        #   fhir_type: {(value, system): fhir_id, ...},
+        #   fhir_type: {(value, system): fhir_id, ...},
+        #   ...
+        # }
+        self.indentifiers_store = {}
+
+        # map from resource_id to resource type
+        self.map_resource_type = {}
+
+        # map_resource_references is a dict of the form
+        # {resource_id: [reference_path, ...], ...}
```

Comment:
can't we remember sub-objects directly rather than paths here ? I want to avoid `find_sub_fhir_object` if possible lol

Reply:
That would mean storing a small dict for each instance that went through the ETL instead of a few strings for each resource...

Reply:
a small dict of less than a 100 bytes right (just a reference) ? I think it's best to have a simple and readable code rather than an optimized one for now

Reply:
Yep but for every instance